src/task.py

```diff
@@ -30,6 +30,11 @@ async def check_announcements(context: ContextTypes.DEFAULT_TYPE) -> None:
             logger.exception(message)
             await context.bot.send_message(chat_id=LOGGER_CHAT_ID, text=message, disable_notification=True)
             continue
+        except ConnectionTimeoutError:
```

Comment:
This doesn't work as intended. Interestingly, `aiohttp` doesn't raise the same error when you set the timeout yourself. This time it is `asyncio.exceptions.TimeoutError`. I did some digging on web, but couldn't figure out why.

I never liked double [inheritance](https://github.com/aio-libs/aiohttp/blob/d58d2c3d0f8755dcd982071e3dc12e9f14db55f6/aiohttp/client_exceptions.py#L232).

Anyway, using `asyncio` one works for us but let's keep aiohttp one just in case. You should catch them together.

Reply:
is it ok if i catch both exceptions in the same block to avoid redundancy? Like this
```python
except (aiohttp.ClientConnectionError, asyncio.exceptions.TimeoutError):
    message = f"Connection Timeout while scraping {department.id}"
    logger.exception(message)
    await context.bot.send_message(chat_id=LOGGER_CHAT_ID, text=message, disable_notification=True)
    continue
```

Reply:
Absolutely!

Reply:
ok, just committed